ecs-agent/wsclient/client_test.go

```diff
@@ -422,3 +445,33 @@ func TestCtxCancel(t *testing.T) {
 	cancel()
 	assert.EqualError(t, <-messageError, "context canceled")
 }
+
+func TestPeriodicDisconnect(t *testing.T) {
+	closeWS := make(chan []byte)
+	defer close(closeWS)
+
+	mockServer, _, _, errChan, _ := utils.GetMockServer(closeWS)
+	mockServer.StartTLS()
+
+	messageError := make(chan error)
+	ctx := context.Background()
+	types := []interface{}{ecsacs.AckRequest{}}
+
+	// Setting a higher rwtimeout to allow disconnect due to periodic timeouts.
+	cs := getTestClientServer(mockServer.URL, types, 20)
+	// Setting up a lower disconnect timer value for testing.
+	timer, err := cs.Connect(mockDisconnectTimeoutMetricName, 10*time.Second, 2*time.Second)
+	require.NoError(t, err)
+	defer timer.Stop()
+	assert.True(t, cs.IsReady(), "expected websocket connection to be ready")
+
+	go func() {
+		messageError <- cs.ConsumeMessages(ctx)
+		cs.Close()
+	}()
+
```

Comment:
I'm thinking we could probably make this test stronger by starting a timer here, stopping the timer once we've received a message on the `errChan` and `messageError` channels, and then asserting that the duration of the timer >= max disconnect time. 

Currently the test asserts that the connection was closed but would still pass even if it was closed right away (instead of in 10-12 seconds which is what we should expect).